src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+                });
+    };
+
+    const onRebaseClicked = () => {
+        onChangeRemoteOrigin(selectedRemote)
+                .then(() => onChangeBranch(selectedBranch))
+                .then(Dialogs.close())
+                .catch(ex => setError(ex.message));
+    };
+
+    const actions = [
+        <Button key="rebase"
+            variant="primary"
+            onClick={onRebaseClicked}
+        >
+            {_("Rebase")}
+        </Button>,
+        <Button key="cancel"
+            variant="link"
+            onClick={Dialogs.close}
+        >
+            {_("Cancel")}
+        </Button>
+    ];
+
+    const repositoryComponent = availableRemotes.length > 1
+        ? (
+            <Select isOpen={isSelectRemoteExpanded}
+                selections={selectedRemote}
+                onToggle={(_, expanded) => setSelectRemoteExpanded(expanded) }
+                onSelect={(_, remote) => { setSelectRemoteExpanded(false); handeRemoteSelect(remote) }}
+                menuAppendTo="parent"
+            >
+                {availableRemotes.map(remote => <SelectOption id={remote} value={remote} key={remote}> {remote} </SelectOption>)}
+            </Select>
+        )
+        : (
+            <Text>{availableRemotes[0]}</Text>
+        );
+
+    const branchComponent = branchLoadError
+        ? (
+            <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }} className="pf-v5-u-danger-color-200">
+                <FlexItem><ExclamationCircleIcon /></FlexItem>
+                <FlexItem>{branchLoadError.replace("error: ", "")}</FlexItem>
+            </Flex>
+        )
+        : (
+            availableBranches.length > 1
+                ? (
+                    <Select isOpen={isSelectBranchExpanded}
+                        selections={selectedBranch}
+                        onToggle={(_, expanded) => setSelectBranchExpanded(expanded) }
+                        onSelect={(_, branch) => { setSelectBranchExpanded(false); setSelectedBranch(branch) }}
+                        menuAppendTo="parent"
+                    >
+                        {availableBranches.map(branch => <SelectOption value={branch} key={branch}> {branch} </SelectOption>)}
+                    </Select>
+                )
+                : (
+                    <Text>{availableBranches[0]}</Text>
+                )
+        );
+
+    const loadingComponent = (
+        <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }}>
+            <FlexItem><Spinner size="md" /></FlexItem>
+            <FlexItem>{_("Loading branches")}</FlexItem>
+        </Flex>
+    );
+
+    return (
+        <Modal title={_("Rebase repository and branch")}
+            position="top"
+            variant="small"
+            id="rebase-repository-modal"
+            isOpen
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            {error && <Alert variant="danger" isInline title={error} />}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231124-162305-47d51597-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)